iceoryx_examples/automotive_soa/include/owl/kom/sample_allocatee_ptr.hpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_SAMPLE_ALLOCATEE_PTR_HPP
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_SAMPLE_ALLOCATEE_PTR_HPP
+
+#include "iceoryx_posh/popo/sample.hpp"
+
+namespace owl
+{
+namespace kom
+{
+/// @brief This class mimics a std::unique_ptr
+template <typename SampleType>
+class SampleAllocateePtr : private iox::cxx::optional<iox::popo::Sample<SampleType>>
+{
+  public:
+    using ParentType = iox::cxx::optional<iox::popo::Sample<SampleType>>;
+
+    using ParentType::ParentType;
```

Comment:
Weird usage, this should cause shadowing. This is the value type of the optional unless I am mistaken but do we need it? It is not the `ParentType` and we should assign another alias like `ValueType`.

Reply:
This does not cause shadowing because we use the scope resolution operator. We need it to make the c'tor public so that users can call it.